src/db/util/raw-model.ts

```diff
@@ -229,6 +266,25 @@ export const defineRawModel =
       await builder.truncate();
     };
 
+    const count: CountFn<F> = async ({ where, trx, distinct } = {}) => {
+      const builder = trx ? masterTable().transacting(trx) : replicaTable();
+      const query = builder.where(prepareCondition(where ?? {}));
+
+      if (distinct !== undefined) {
+        query.countDistinct(distinct);
+      } else {
+        query.count('*');
+      }
+
+      const res = (await query) as unknown as Array<{ count: string }>;
```

Comment:
In my [suggested change](https://github.com/UN-OCHA/hpc-api-core/pull/152#discussion_r1674313558) from previous round of reviews, I have suggested using
```ts
const res = (await query) as unknown as [{ count: string }];
```
but you defined it using `Array`. I want to point out the difference. I have suggested using a **tuple**, which specifies this as an **array of single element**, which has object with `count` property. When you use `Array`, you indicate there can be multiple objects in an array, each having `count` property. You can see the distinction if you hover `res.length` on line 281, TS says that its type is `number`. If you change the code to use a tuple, as I suggested, the type will be `1`.

In this case, typing it as a tuple is more realistic, because that is what Knex returns, but it doesn't bring any difference to the code, as this is purely typing thing. Just wanted to explain the difference, you may not want to use a tuple for some reason.